### ISSUE TITLE
ANSI Color codes break the CLI integration

### ISSUE THREAD

**Issue by vlad-acertus** [forcedotcom/salesforcedx-vscode#5695]
### Summary

The extension breaks because ANSI color codes are included in the SF CLI results in the stdout stream.

### Steps To Reproduce:

1.  Have Ubuntu with ANSI colors enabled in terminal ( .profile or .bashrc with FORCE_COLOR=true env)
2. Install VS Code
3. Install SFDX CLI
4. Install "Salesforce Extension Pack" 
5. Run Authorize and Org and get "success" message
7.  Run any SF CLI command via the VS Code command prompt or click refresh button in the org explorer of the Salesforce Extension Pack

### Expected result

Shows contents of the org in the org explorer when refreshing the org explorer after authorizing an org.

### Actual result

Received an error message "Error fetching metadata for org. Run "SFDX: Authorize an Org" to authorize your org again."

### Additional information

Here is the code that is making the problem:
![image](https://github.com/user-attachments/assets/98e502a6-2725-4450-a835-98aab483cc1e)
Getting error when trying to refresh the org browser:
![image](https://github.com/user-attachments/assets/befaea61-7242-4bff-8adb-602ea52f30c2)

Quick hack to demonstrate the issue and solution (removing the ANSI color codes from the output string):
![image](https://github.com/user-attachments/assets/f732a37a-6e09-4d91-89b7-cd317c3dbfe5)

Refresh org browser after adding the fix:
![image](https://github.com/user-attachments/assets/7eed1fa6-c914-4780-ba45-efbc3a46bd74)

I would not recommend fixing this by clearing the string, instead I would try to disable any ANSI colors in the shell when running the SFDX CLI command. This should fix all problems instead of patching it down stream.

**Salesforce Extension Version in VS Code**: 61.5.0

**Salesforce CLI Version**: @salesforce/cli/2.50.6 linux-x64 node-v20.9.0

**OS and version**: Ubuntu 20.04.6 LTS

**VS Code version**: 1.91.1

**Most recent version of the extensions where this was working**: Not sure

**Comment by vlad-acertus** [comment]
The version numbers are there, I will update them to match the format you are looking for.

Solution: Before running any CLI commands export "NO_COLOR" env. SF cli supports it.
![image](https://github.com/user-attachments/assets/597549ac-734a-433f-bc6f-39401cd7e78f)

**Comment by mingxuanzhangsfdx** [comment]
Hi @vlad-acertus , thanks for reporting the error. I checked your use case on my Mac and it works fine. I will try to reproduce it on ubuntu and let you know.

**Comment by vlad-acertus** [comment]
> Hi @vlad-acertus , thanks for reporting the error. I checked your use case on my Mac and it works fine. I will try to reproduce it on ubuntu and let you know.

I did some modification of the error handler to return the raw error message. When I run the command I get this result:
![image](https://github.com/user-attachments/assets/937d55b8-d421-48a9-8ee2-5fb7641897b3)

I did some digging and found that "FORCE_COLOR" env causes the error to appear. This is the result of SFDX CLI being compliant with https://force-color.org/. I have the "FORCE_COLOR" env enabled on my machine because it forces the colors to appear in the terminal for all cli tools which makes it much easier to read than black and white text. 

The current work around I have is to keep the string replace functionality in the minified transpiled js code and continuously update it each time the extension updates.

To fix this completely, the "FORCE_COLOR" env should be removed in the shell when executing the SFDX cli command from the extension and also enabling "NO_COLOR" env so that the SFDX cli doesn't return ANSI color codes in the stdout stream.

**Comment by mingxuanzhangsfdx** [comment]
Hi @vlad-acertus , I managed to set up an ubuntu and follow the steps to reproduce. 

For the step 1, 

In the file ~/.bashrc, 

1. I enable the line `force_color_prompt=yes` 
2. I set the `if` block 

```
# enable color support of ls and also add handy aliases
if [ -x /usr/bin/dircolors ]; then
    test -r ~/.dircolors && eval "$(dircolors -b ~/.dircolors)" || eval "$(dircolors -b)"
    alias ls='ls --color=auto'
    alias dir='dir --color=auto'
    alias vdir='vdir --color=auto'

    alias grep='grep --color=auto'
    alias fgrep='fgrep --color=auto'
    alias egrep='egrep --color=auto'
fi
```
3. I added a new line `FORCE_COLOR=1`

4. and I ran `source ~/.bashrc` to load the config. 

After that, I verified I got them enabled

<img width="768" alt="image" src="https://github.com/user-attachments/assets/2c1e6904-b297-41a2-a02f-eb62a7e0615d">

I tried to run several cli integration commands such as Refresh SObjects and refresh org browser as you suggested but saw nothing wrong. 

Please let me know how I can reproduce it.

**Comment by daphne-sfdc** [comment]
Hi @vlad-acertus, We have released v61.8.1 of the Salesforce Extensions for VSCode today, which contains the fix for your issue.  Therefore, I'm going to close the issue.
